### PR TITLE
Fix how we invalidate the sbt bridge cache

### PR DESCRIPTION
**Pull request by smarter**
- Remove the correct files to actually get rid of the cache
- Only clean the cache when the bridge has been recompiled
- Make sure the bridge is compiled when compiling a bootstrapped project

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -746,18 +748,29 @@ object Build {
   // until sbt/sbt#2402 is fixed (https://github.com/sbt/sbt/issues/2402)
   lazy val cleanSbtBridge = TaskKey[Unit]("cleanSbtBridge", "delete dotty-sbt-bridge cache")
 
+  def cleanSbtBridgeImpl(): Unit = {
+    val home = System.getProperty("user.home")
+    val sbtOrg = "org.scala-sbt"
+    val bridgePattern = s"*dotty-sbt-bridge*$dottyVersion*"
+
+    IO.delete((file(home) / ".sbt" / "1.0" / "zinc" / sbtOrg * bridgePattern).get)
+    IO.delete((file(home) / ".sbt"  / "boot" * "scala-*" / sbtOrg / "sbt" * "*" * bridgePattern).get)
```
Comment (Blaisorblade) [inline project/Build.scala]: How safe is deleting wildcard files under the home directory? Looks a bit scary (both the code and especially that it's allowed). Tho apparently we already did it?

Comment (smarter) [reply project/Build.scala]: The paths used are as precise as I can reasonably make them to limit the use of wildcards. Suggestions on making this better are welcome but I think it's fine as is.

**Comment by smarter** [comment]
Yeah I don't think it's worth it

**Comment by smarter** [comment]
What would be useful is invalidating bootstrapped projects when the tasty major version is bumped but I don't know how to do that

**Comment by jvican** [comment]
FWIW, using Zinc's scripted infrastructure this wouldn't be required, as the compiler bridges are sourced from the classes directory of the sbt projects.

**Pull request merged**